inspire/base/static/js/search/search_results.js

```diff
@@ -44,8 +44,215 @@ require(['jquery', 'bootstrap'], function($) {
+  });
+
+  $("#export-as-cv-html").on('click', function() {
+    if (sList != 0) {
+      create_url('/formatter/export-as/cv_latex_html/', sList);
+    } else {
+      $("#download").removeAttr('href');
+    }
+    $("#dropdown-export").html('CV format (html) <span class="caret"></span>');
+  });
+
+  $("#export-as-cv-text").on('click', function() {
+    if (sList != 0) {
+      create_url('/formatter/export-as/cv_latex_text/', sList);
+    } else {
+      $("#download").removeAttr('href');
+    }
+    $("#dropdown-export").html('CV format (text) <span class="caret"></span>');
+  });
+
+  var sList = [];
+  $('#checkbox-select-all').click(function(event) {
+    if (this.checked) {
+      $('.checkbox-results').each(function() {
+        this.checked = true;
+        sList.push($(this).attr("id"));
+      });
+      $('#results-panel').after('<div class="panel panel-default" id="info-message">' +
+        '<div class="panel-body" >You have selected ' + sList.length + ' records of this page. <a class="pointer" id="select-all-records">' +
+        'Select all ' + $('#total-results').text() + ' results.</a></div></div>');
+      $('#select-all-records').on('click', function() {
+        $('#info-message').remove();
+        $('#results-panel').after('<div class="alert alert-warning" id="alert-selection" role="alert">' +
+          $('#total-results').text() + ' have been selected.<a class="pointer" id="undo-selection"> Undo selection.</a></div>');
+        $.get("/search?of=id&rg=" + $('#total-results').text(), function(data, status) {
+          if (status == 'success') {
+            sList = data;
+            create_url(getType(), sList);
+          } else {
+            sList = [];
+            $('#alert-selection').remove();
+            $("#download").removeAttr('href');
+            $('#results-panel').after('<div class="alert alert-danger" id="danger-selection" role="alert">' +
+              'Too many records to download.</div>')
+          }
+        });
+      });
+    } else {
+      $('.checkbox-results').each(function() {
+        this.checked = false;
+      });
+      sList = [];
+      $("#download").removeAttr('href');
+      $('#info-message').remove();
+      $('#alert-selection').remove();
+    }
+  });
+
+  $(document).on('click', '#undo-selection', function(e) {
+    sList = [];
+    $('#alert-selection').remove();
+    $("#download").removeAttr('href');
+    $('.checkbox-results').each(function() {
+      this.checked = false;
+    });
+    $('#checkbox-select-all').prop('checked', false);
+  });
+
+  function querystring(key) {
+    var re = new RegExp('(?:\\?|&)' + key + '=(.*?)(?=&|$)', 'gi');
+    var r = [],
+      m;
+    while ((m = re.exec(document.location.search)) != null) r.push(m[1]);
+    return r;
+  }
+
+  $('input[type=checkbox]').change(function() {
+    var $this = $(this);
+    if (this.checked) {
+      if ($this.attr("id") != 'checkbox-select-all') {
```

Comment:
This will not be needed if the selector only matches the checkboxes close to each result.